### PR TITLE
README:  document package localisation

### PR DESCRIPTION
**Pull request by deepfire**
In the README:

1. Add a ToC.
2. Fix the structure, so the ToC actually makes sense.
3. Fix some minor RST syntax issues.
4. Some very minor fixes.
5. Expand the Development section with the workbench usage info.
6. Add `cabal-plan` and `graphmod` to the workbench shell.
7. Makes the node start on Darwin, when metric collection is enabled, by fixing https://github.com/input-output-hk/cardano-node/issues/4380

### CODE REVIEW

**Review by newhoggy — APPROVED** [review]

**Comment by newhoggy** [comment]
I'm interested to know if the new instructions will work on a Mac (Intel or M1).  I think this should be documented to let people know whether the instructions are worth trying on these platforms.

File: README.rst [hunk 1]
```diff
@@ -413,54 +412,183 @@ Byron vote submission:
...
+
+By default, all binaries originating in the ``cardano-node`` repository are available to ``cabal build`` and ``cabal run``, unless the workbench was entered using one of the pure ``*nix`` modes.  Note that in all cases, the dependencies for the workbench are supplied though Nix and have been built/tested on CI.
+
+**Dependency localisation** -or- *Cabal&Nix for painless cross-repository work*
+----
+
+Normally, to work on ``cardano-node`` dependencies in the context of the node itself, one needs to go through an expensive multi-step process, mandated by the Nix model:
+    - Make changes to a dependency (say ``ouroboros-consensus-cardano`` in the ``ouroboros-network`` repository), commit them & push to Github.
+    - Calculate & bump the pin for that dependency in ``cardano-node``, inside its ``cabal.project``.
+    - Re-enter the Nix shell in ``cardano-node``, suffering full rebuilds of everything.
```
Comment (ch1bo) [inline README.rst:461]: What's the point of describing this workflow if the actual thing you want to document is below? Also, the language used here is sarcastic and makes the reader question why we are using `nix` if it is resulting in "suffering full rebuilds of everything".

I would suggest to frame the whole section of "Dependency localisation" in a more positive way.

Comment (deepfire) [reply README.rst:461]: > Also, the language used here is sarcastic and makes the reader question why we are using nix if it is resulting in "suffering full rebuilds of everything".

Was not intended to be sarcastic at all -- just recognising a painful reality we all had to deal with.

But perhaps you are right -- I should find a more neutral wording.

**Review by jutaro — APPROVED** [review]

**Comment by mgmeier** [comment]
> I'm interested to know if the new instructions will work on a Mac (Intel or M1). I think this should be documented to let people know whether the instructions are worth trying on these platforms.

I've tested both `cabal-plan` and `graphmod` build inputs for workbench successfully on an Intel Mac.

**Review by mgmeier — APPROVED** [review]

**Pull request merged**